src/learning/LinearSolver.cpp

```diff
@@ -0,0 +1,225 @@
+#include "../../include/learning/LinearSolver.h"
+#include "../../include/app.h"
+#include "../../include/types/AlternativesPerformance.h"
+#include "ortools/linear_solver/linear_solver.h"
+
+#include <sstream>
+#include <string>
+
+LinearSolver::LinearSolver(AlternativesPerformance &ap, Config &conf,
+                           float delta, std::string solver_name)
+    : ap(ap), conf(conf) {
+  solver = operations_research::MPSolver::CreateSolver(solver_name);
+  solver_name = solver_name;
+  delta = delta;
+}
+
+LinearSolver::~LinearSolver() { delete solver; }
+
+AlternativesPerformance &LinearSolver::getAlternativesPerformance() const {
+  return ap;
+}
+
+operations_research::MPSolver *LinearSolver::getSolver() const {
+  return solver;
+}
+
+Config LinearSolver::getConf() const { return conf; }
+
+std::vector<operations_research::MPVariable *>
+LinearSolver::getWeights() const {
+  return weights;
+}
+
+std::vector<operations_research::MPVariable *> LinearSolver::getXa() const {
+  return x_a;
+}
+
+std::vector<operations_research::MPVariable *> LinearSolver::getXap() const {
+  return x_ap;
+}
+
+std::vector<operations_research::MPVariable *> LinearSolver::getYa() const {
+  return y_a;
+}
+
+std::vector<operations_research::MPVariable *> LinearSolver::getYap() const {
+  return y_ap;
+}
+
+void LinearSolver::initializeSolver() {
+
+  // reset constraints vectors
+  x_constraints.clear();
+  y_constraints.clear();
+  weights_constraint.clear();
+
+  // reset variables vectors
+  x_a.clear();
+  x_ap.clear();
+  y_a.clear();
+  y_ap.clear();
+  weights.clear();
+
+  // reset previous solver
+  solver->Clear();
```

Comment:
why is this one capitalized? (C maj) 
I'm just curious

Reply:
This is the convention used by ortools, `solver->Clear` is not something we implemented ourself